app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js

```diff
@@ -0,0 +1,534 @@
+                aNode = document.createElement('a');
+                aNode.setAttribute('href', gitURLs[1]);
+                aNode.appendChild(document.createTextNode(gitCommit));
+                aNode.insertAdjacentHTML('beforeend', '&nbsp;');
+                aNode.appendChild(html.external());
+            } else {
+                if (gitCommit && gitCommit !== null) {
+                    aNode = document.createTextNode(gitCommit);
+                } else {
+                    aNode = html.nonavail();
+                }
+            }
+            html.replaceContent(document.getElementById('git-commit'), aNode);
+
+            // Date.
+            domNode = document.createElement('time');
+            domNode.setAttribute('datetime', createdOn.toISOString());
+            domNode.appendChild(
+                document.createTextNode(createdOn.toCustomISODate()));
+            html.replaceContent(
+                document.getElementById('job-date'), domNode);
+        }
+    }
+
+    function getTests() {
+        var data;
+        var deferred;
+
+        function getData(lab) {
+            data.lab_name = lab;
+
+		    gLabTable.push(table({
+		        tableId: 'lab-table-' + lab,
+		        tableLoadingDivId: 'lab-table-loading-' + lab,
+		        tableDivId: 'lab-table-div-' + lab
+		    }));
+		    var labTable = gLabTable[gLabTable.length - 1];
+            deferred = request.get('/_ajax/test/suite', data);
+            $.when(deferred)
+                .done(function(response) {
+                    getTestsDone(lab, labTable, response);
+                    getBatchCount(lab, labTable, response);
+                })
+                .fail(error.error, function() {
+                    getTestsFail(lab);
+                });
+        }
+
+        data = {
+            board: gBoard,
+            job: gJob,
+            kernel: gKernel,
+            sort: 'created_on',
+            sort_order: '-1',
+        };
+        // Update the general details
+        deferred = request.get('/_ajax/test/suite', data);
+        $.when(deferred)
+            // .fail(error.error, )
+            .done(updateDetails)
+        // Update each lab with the specific data
+        gLabList.forEach(getData);
+    }
+
+    if (document.getElementById('job-name') !== null) {
+        gJob = document.getElementById('job-name').value;
+    }
+    if (document.getElementById('kernel-name') !== null) {
+        gKernel = document.getElementById('kernel-name').value;
+    }
+    if (document.getElementById('board-name') !== null) {
+        gBoard = document.getElementById('board-name').value;
+    }
+    if (document.getElementById('search-filter') !== null) {
+        gSearchFilter = document.getElementById('search-filter').value;
+    }
+    if (document.getElementById('file-server') !== null) {
+        gFileServer = document.getElementById('file-server').value;
+    }
+
+    gLabList = document.getElementById('lab-names');
```

Comment:
Hmmm... probably, in this case instead of going through a `split` call, it might be better to add multiple `input` elements to the HTML page, retrieve them via `querySelectorAll` and then apply a `map` function to extract the values (you actually need an extra step to convert what `querySelectorAll` returns, since it's not an Array). It would be something like:
```
Array.prototype.slice.call(document.querySelectorAll('input.lab-names')).map( ... );
```

In case this needs to be changed in the HTML template to add the multiple `input`s with a `class` name of `lab-names` (or something else).